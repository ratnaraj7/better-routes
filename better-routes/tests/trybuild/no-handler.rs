use axum::response::{IntoResponse, Response};
use axum_extra::routing::RouterExt;
use better_routes::{method_helper, routes};

#[derive(Default)]
struct GlobalRejection;

impl IntoResponse for GlobalRejection {
    fn into_response(self) -> Response {
        todo!()
    }
}

#[derive(Default)]
struct FooRejection;

impl IntoResponse for FooRejection {
    fn into_response(self) -> Response {
        todo!()
    }
}

#[derive(Clone)]
struct State {}

routes! {
    State => State,
    Rejection => GlobalRejection,
    "/foo" => struct Foo; => FooRejection,
    "/bar" => struct Bar;,
}

#[method_helper(State)]
impl Foo {}

#[method_helper(State)]
impl Bar {
    #[get]
    async fn bar(self) {}
}

fn main() {}
