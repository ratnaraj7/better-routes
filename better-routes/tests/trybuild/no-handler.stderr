error: No methods found
  --> tests/trybuild/no-handler.rs:34:1
   |
34 | impl Foo {}
   | ^^^^

error[E0277]: the trait bound `Foo: MethodHandler<State>` is not satisfied
  --> tests/trybuild/no-handler.rs:29:22
   |
29 |     "/foo" => struct Foo; => FooRejection,
   |                      ^^^ the trait `MethodHandler<State>` is not implemented for `Foo`
   |
   = help: the trait `MethodHandler<State>` is implemented for `Bar`
note: required by a bound in `get_all_routes`
  --> tests/trybuild/no-handler.rs:26:1
   |
26 | / routes! {
27 | |     State => State,
28 | |     Rejection => GlobalRejection,
29 | |     "/foo" => struct Foo; => FooRejection,
30 | |     "/bar" => struct Bar;,
31 | | }
   | |_^ required by this bound in `get_all_routes`
   = note: this error originates in the macro `routes` (in Nightly builds, run with -Z macro-backtrace for more info)
